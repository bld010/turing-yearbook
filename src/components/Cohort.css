.Cohort {
  display: flex;
  flex-wrap: wrap
}