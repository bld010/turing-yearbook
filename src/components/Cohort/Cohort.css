
div .Cohort {
  display: flex;
  flex-wrap: wrap;
  margin: auto;
  justify-content: left;
}

.Cohort h2 {
  display: block;
  text-align: left; 
  font-weight: 300;
  text-transform: uppercase;
}