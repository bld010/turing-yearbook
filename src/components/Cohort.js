import React from 'react';
import Person from './Person';
import './Cohort.css';

const Cohort = ({ people, updatePeople, cohortType }) => {
  return (
    <div className="Cohort">
      {people.map(person => {
      return <Person 
        key={person.name}  
        cohortType={cohortType} 
        updatePeople={updatePeople}
        person={person} />
      })}
    </div>
  )
  
}

export default Cohort;
