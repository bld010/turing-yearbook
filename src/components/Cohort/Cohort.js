import React from 'react';
import Person from '../Person/Person';
import './Cohort.css';

const Cohort = ({ people, updatePeople, cohortType, deletePerson }) => {
  return (
    <section className="Cohort">
      <h2>{cohortType}</h2>
      <div className={cohortType + " Cohort"} >
        {people.map(person => {
        return <Person 
          key={person.name}  
          cohortType={cohortType} 
          updatePeople={updatePeople}
          deletePerson={deletePerson}
          person={person} />
        })}
      </div>
    </section>
  )
  
}

export default Cohort;
