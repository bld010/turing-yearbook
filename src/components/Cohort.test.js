import React from 'react';
import { shallow } from 'enzyme';
import Cohort from './Cohort';
import people from '../data/yearbook-data'

it('should match the snapshot with all correct data', () => {
  const wrapper = shallow(<Cohort people={people.staff} cohortType="staff" deletePerson={jest.fn()} updatePeople={jest.fn()}/>)

  expect(wrapper).toMatchSnapshot();
});

