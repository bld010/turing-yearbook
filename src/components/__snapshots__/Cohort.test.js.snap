// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot with all correct data 1`] = `
<div
  className="Cohort"
>
  <Person
    cohortType="staff"
    deletePerson={[MockFunction]}
    key="Eric"
    person={
      Object {
        "id": 1,
        "name": "Eric",
        "photo": "./eric.jpg",
        "quote": "You miss all the shots you don't take.",
        "superlative": "Most Likely to Win the PGA",
      }
    }
    updatePeople={[MockFunction]}
  />
  <Person
    cohortType="staff"
    deletePerson={[MockFunction]}
    key="Amy"
    person={
      Object {
        "id": 2,
        "name": "Amy",
        "photo": "./amy.jpg",
        "quote": "I believe in you, cuties!",
        "superlative": "Most Likely to Be an Actual Superhero",
      }
    }
    updatePeople={[MockFunction]}
  />
  <Person
    cohortType="staff"
    deletePerson={[MockFunction]}
    key="Khalid"
    person={
      Object {
        "id": 3,
        "name": "Khalid",
        "photo": "./khalid.png",
        "quote": "Don't put the world on your shoulders.",
        "superlative": "Most Likely to Save the Planet from Alien Invasion",
      }
    }
    updatePeople={[MockFunction]}
  />
  <Person
    cohortType="staff"
    deletePerson={[MockFunction]}
    key="Travis"
    person={
      Object {
        "id": 4,
        "name": "Travis",
        "photo": "./travis.jpg",
        "quote": "You do you.",
        "superlative": "Most Likely to Win a Grammy",
      }
    }
    updatePeople={[MockFunction]}
  />
  <Person
    cohortType="staff"
    deletePerson={[MockFunction]}
    key="Leta"
    person={
      Object {
        "id": 5,
        "name": "Leta",
        "photo": "./leta.jpg",
        "quote": "It's okay to be kind of a mess.",
        "superlative": "Most Likely to Go to the Moon",
      }
    }
    updatePeople={[MockFunction]}
  />
  <Person
    cohortType="staff"
    deletePerson={[MockFunction]}
    key="Christie"
    person={
      Object {
        "id": 6,
        "name": "Christie",
        "photo": "./christie.jpg",
        "quote": "Every journey begins with a single step. So take the damn step!",
        "superlative": "Most Likely to Be President",
      }
    }
    updatePeople={[MockFunction]}
  />
  <Person
    cohortType="staff"
    deletePerson={[MockFunction]}
    key="Brittany"
    person={
      Object {
        "id": 7,
        "name": "Brittany",
        "photo": "./brittany.jpg",
        "quote": "It was Robbie.",
        "superlative": "Most Likely to Get Away with Murder",
      }
    }
    updatePeople={[MockFunction]}
  />
  <Person
    cohortType="staff"
    deletePerson={[MockFunction]}
    key="Robbie"
    person={
      Object {
        "id": 8,
        "name": "Robbie",
        "photo": "./robbie.jpg",
        "quote": "Cool.",
        "superlative": "Most Likely to Be Accused of Murder",
      }
    }
    updatePeople={[MockFunction]}
  />
  <Person
    cohortType="staff"
    deletePerson={[MockFunction]}
    key="Will"
    person={
      Object {
        "id": 9,
        "name": "Will",
        "photo": "./will.jpg",
        "quote": "Stop worrying and just do the thing.",
        "superlative": "Most Likely to Invent Cold Fusion",
      }
    }
    updatePeople={[MockFunction]}
  />
  <Person
    cohortType="staff"
    deletePerson={[MockFunction]}
    key="Pam"
    person={
      Object {
        "id": 10,
        "name": "Pam",
        "photo": "./pam.jpg",
        "quote": "Shit!",
        "superlative": "Most Likely to Win a Gold Medal",
      }
    }
    updatePeople={[MockFunction]}
  />
  <Person
    cohortType="staff"
    deletePerson={[MockFunction]}
    key="David"
    person={
      Object {
        "id": 11,
        "name": "David",
        "photo": "./david.JPG",
        "quote": "I love git.",
        "superlative": "Most Likely to Be Eaten by a Shark",
      }
    }
    updatePeople={[MockFunction]}
  />
  <Person
    cohortType="staff"
    deletePerson={[MockFunction]}
    key="Louisa"
    person={
      Object {
        "id": 12,
        "name": "Louisa",
        "photo": "./louisa.jpg",
        "quote": "YOLO",
        "superlative": "Most Likely to Win a Knife Fight",
      }
    }
    updatePeople={[MockFunction]}
  />
  <Person
    cohortType="staff"
    deletePerson={[MockFunction]}
    key="Jeff"
    person={
      Object {
        "id": 13,
        "name": "Jeff",
        "photo": "https://i.ytimg.com/vi/DhqzMc_LXgQ/maxresdefault.jpg",
        "quote": "::thoughtful nonsense::",
        "superlative": "Most Likely to Tell a Dad Joke",
      }
    }
    updatePeople={[MockFunction]}
  />
</div>
`;
